### ISSUE TITLE
FastClasspathScanner should support Java EA Version String

### ISSUE THREAD

**Issue by miho** [lukehutch/fast-classpath-scanner#214]
Currently, FastClasspathScanner 3 doesn't seem to work with EA versions of Java simply because it tries to parse the version String as number (see **`Caused by: java.lang.NumberFormatException: For input string: "11-ea"`**). This is unfortunate, since it prevents users from doing testing against EA versions. Since Java has a much more frequent release cycle testing with EA releases is super important. Are you aware of the issue? Thanks!

```
Caused by: java.lang.RuntimeException: java.lang.ExceptionInInitializerError
        at io.github.lukehutch.fastclasspathscanner.FastClasspathScanner.scan(FastClasspathScanner.java:1640)
        at io.github.lukehutch.fastclasspathscanner.FastClasspathScanner.scan(FastClasspathScanner.java:1674)
        at io.github.lukehutch.fastclasspathscanner.FastClasspathScanner.scan(FastClasspathScanner.java:1700)
        at eu.mihosoft.vmf.VMF.listClassesInPackage(VMF.java:128)
        at eu.mihosoft.vmf.VMF.generate(VMF.java:114)
        at eu.mihosoft.vmf.VMF$generate.call(Unknown Source)
        at eu.mihosoft.vmf.gradle.plugin.VMFPlugin$_apply_closure5$_closure12.doCall(VMFPlugin.groovy:110)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:732)
        at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:705)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:131)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:317)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:309)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:185)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:97)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:120)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:99)
        ... 105 more
Caused by: java.lang.ExceptionInInitializerError
        at io.github.lukehutch.fastclasspathscanner.scanner.ModuleRef.findModuleRefs(ModuleRef.java:423)
        at io.github.lukehutch.fastclasspathscanner.scanner.ClassLoaderAndModuleFinder.<init>(ClassLoaderAndModuleFinder.java:154)
        at io.github.lukehutch.fastclasspathscanner.scanner.ClasspathFinder.<init>(ClasspathFinder.java:190)
        at io.github.lukehutch.fastclasspathscanner.scanner.Scanner.call(Scanner.java:214)
        at io.github.lukehutch.fastclasspathscanner.scanner.Scanner.call(Scanner.java:55)
Caused by: java.lang.NumberFormatException: For input string: "11-ea"
        at io.github.lukehutch.fastclasspathscanner.utils.ReflectionUtils.<clinit>(ReflectionUtils.java:313)
```

**Comment by miho** [comment]
Seems to be fixed in **3.1.12**. Sorry for the noise.

**Comment by lukehutch** [comment]
@miho Yes, this was reported once before, and fixed, but thanks for reporting!

Just as an unrelated heads-up, I am mostly finished with a huge refactoring and API cleanup effort, which will be released as v4 in the next couple of days, so check back soon for a much improved (and different) API.
